R/combinedscatterplot.R

```diff
@@ -0,0 +1,827 @@
+                            x.tick.show = TRUE,
+                            x.tick.suffix = "",
+                            x.tick.prefix = "",
+                            x.tick.format = "",
+                            x.hovertext.format = "",
+                            x.tick.angle = NULL,
+                            x.tick.font.color = global.font.color,
+                            x.tick.font.family = global.font.family,
+                            x.tick.font.size = 10,
+                            x.tick.label.wrap = TRUE,
+                            x.tick.label.wrap.nchar = 21,
+                            hovertext.font.family = global.font.family,
+                            hovertext.font.size = 11,
+                            marker.size = 6,
+                            swap.x.and.y = FALSE,
+                            legend.bubbles.show = TRUE,
+                            label.auto.placement = TRUE)
+{
+    orig.x <- x
+    checkDataIsEnough(x, y)
+
+    # Try to store name of variables
+    scatter.mult.yvals <- isTRUE(attr(x, "scatter.mult.yvals"))
+    if (!is.null(scatter.sizes) && is.null(scatter.sizes.name))
+        scatter.sizes.name <- deparse(substitute(scatter.sizes))
+    if (!is.null(scatter.labels) && is.null(scatter.labels.name))
+        scatter.labels.name <- deparse(substitute(scatter.labels))
+    if (!is.null(scatter.colors) && is.null(scatter.colors.name))
+        scatter.colors.name <- deparse(substitute(scatter.colors))
+
+    num.tables <- 1
+    groups <- NULL
+    if (is.list(x) && !is.null(ncol(x[[1]])))
+    {
+        output <- unlistX(x, trend.lines)
+        x <- output$x
+        groups <- output$groups
+        num.tables <- output$num.tables
+    }
+
+    x <- convertPercentToProportion(x)
+    annot.data <- x
+
+    if (is.matrix(x) || is.data.frame(x))
+    {
+        output <- unpackColumnsFromX(x, y, scatter.labels, scatter.x.column,
+                                     scatter.y.column, scatter.mult.yvals,
+                                     x.title, y.title, scatter.sizes,
+                                     scatter.sizes.column, scatter.sizes.name,
+                                     scatter.colors, scatter.colors.column,
+                                     scatter.colors.name)
+        x <- output$x
+        y <- output$y
+        scatter.labels <- output$scatter.labels
+        x.title <- output$x.title
+        y.title <- output$y.title
+        scatter.sizes <- output$scatter.sizes
+        scatter.sizes.name <- output$scatter.sizes.name
+        scatter.colors <- output$scatter.colors
+        scatter.colors.name <- output$scatter.colors.name
+    }
+
+    if (is.null(x) && is.null(y))
+        stop("At least one of x or y must be supplied.")
+
+    if (is.null(x))
+    {
+        x <- rep(0, length(y))
+        if (!any(nzchar(x.bounds.minimum)))
+            x.bounds.minimum = -0.25
```

Comment:
Where do these defaults come from?

Reply:
This is the case where x isn't supplied, so the x data is set to 0, and a small min and max is set around it (not sure if it is necessary to do this as the scatterplot might set similar bounds, I just copied this over from `LabeledScatterplot`).

Reply:
I see. They might not be necessary any more because plotly should be able to null bounds but I guess we can just leave it since its in LabeledScatter